src/btree/bt_discard.c

```diff
@@ -39,6 +39,10 @@ __wt_ref_out(WT_SESSION_IMPL *session, WT_REF *ref)
      */
     WT_ASSERT(session, __wt_hazard_check_assert(session, ref, true));
 
+    WT_ASSERT(session, !WT_PAGE_IS_INTERNAL(ref->page) ||
+        F_ISSET(session->dhandle, WT_DHANDLE_EXCLUSIVE) ||
```

Comment:
This change requires the change in WT-7065 at the same time or the assertion will fail. See WT-7920 which was another backport of this change.